adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -338,56 +334,100 @@ case class ReferenceRegion(
   /**
    * Merges two reference regions that are contiguous.
    *
-   * @throws AssertionError Thrown if regions are not overlapping or adjacent.
+   * @throws IllegalArgumentException Thrown if regions are not overlapping or adjacent.
    *
-   * @param region Other region to merge with this region.
+   * @param other Other region to merge with this region.
    * @return The merger of both unions.
    *
    * @see hull
    */
-  def merge(region: ReferenceRegion): ReferenceRegion = {
-    assert(overlaps(region) || isAdjacent(region), "Cannot merge two regions that do not overlap or are not adjacent")
-    hull(region)
+  def merge(other: ReferenceRegion): ReferenceRegion = {
+    require(overlaps(other) || isAdjacent(other), "Cannot merge two regions that do not overlap or are not adjacent")
+    merge(other, 1L)
+  }
+
+  /**
+   * Merges two reference regions that are within a threshold of each other.
+   *
+   * @throws IllegalArgumentException Thrown if regions are no within the distance threshold.
+   *
+   * @param other Other region to merge with this region.
+   * @return The merger of both unions.
+   *
+   * @see hull
+   */
+  def merge(other: ReferenceRegion, distanceThreshold: Long): ReferenceRegion = {
+    require(isNearby(other, distanceThreshold), "Cannot merge two regions that do not meet the distance threshold")
+    require(distanceThreshold >= 0, "Distance must be non-negative number")
+    hull(other)
   }
 
   /**
    * Calculates the intersection of two reference regions.
    *
-   * @param region Region to intersect with.
+   * @param other Region to intersect with.
    * @return A smaller reference region.
    */
-  def intersection(region: ReferenceRegion): ReferenceRegion = {
-    assert(overlaps(region), "Cannot calculate the intersection of non-overlapping regions.")
-    ReferenceRegion(referenceName, max(start, region.start), min(end, region.end))
+  def intersection(other: ReferenceRegion): ReferenceRegion = {
+    require(overlaps(other), "Cannot calculate the intersection of non-overlapping regions.")
+    ReferenceRegion(referenceName, max(start, other.start), min(end, other.end))
```

Comment:
+1 on the default, and yes good catch. We weren't setting the strand before, but we definitely should be.